frontend/simple/js/transactions.js

```diff
@@ -0,0 +1,263 @@
+'use strict'
+
+import localforage from 'localforage'
+import * as Events from '../../../shared/events'
+import sbp from '../../../shared/sbp'
+
+let _steps = new WeakMap()
+let _scope = new WeakMap()
+let _cursor = new WeakMap()
+let _description = new WeakMap()
+let _isRecoverable = new WeakMap()
+let _wait = new WeakMap()
+// Why would need this?  to make our retries grow in a common sense ways
+function fibSequence (num) {
+  let output = 0
+  let sequence = [ 0, 1 ]
+  for (var i = 0; i < num; i++) {
+    sequence.push(sequence[ 0 ] + sequence[ 1 ])
+    sequence.splice(0, 1)
+    output = sequence[ 1 ]
+  }
+  return output
+}
+
+let transactionDB = localforage.createInstance({ name: 'Transactions', storeName: 'Transactions' })
+export class Transaction {
+  constructor (description, recoverable) {
+    _description.set(this, description)
+    _steps.set(this, [])
+    _scope.set(this, {})
+    _isRecoverable.set(this, !!recoverable)
+    _cursor.set(this, 0)
+    let res
+    let rej
+    let promise = new Promise((resolve, reject) => (res = resolve) && (rej = reject))
+    _wait.set(this, {promise, resolve: res, reject: rej})
+  }
+  get isRecoverable () {
+    return _isRecoverable.get(this)
+  }
+  wait () {
+    let wait = _wait.get(this)
+    return wait.promise
+  }
+  addStep (step) {
+    let steps = _steps.get(this)
+    steps.push(step)
+  }
+  // This function creates variables in the transaction's scope to be referenced by steps
+  setInScope (name, value) {
+    let scope = _scope.get(this)
+    scope[name] = value
+    _scope.set(this, scope)
+  }
+  async run (serviceRegistry) {
```

Comment:
`serviceRegistry` is unused. Why did `standard` not catch that?

Reply:
That's not sure that is something standard should catch you may have unused parameters. I had removed this but had attempted to undo an error in a lower block of code reintroduced it unintentionally.